libminifi/src/core/ProcessGroup.cpp

```diff
@@ -92,13 +92,12 @@ ProcessGroup::~ProcessGroup() {
     onScheduleTimer_->stop();
   }
 
-  for (auto &&connection : connections_) {
+  for (auto&& connection : connections_) {
```

Comment:
That's a forwarding reference, because `auto` uses template deduction rules. In other words, "I don't care what it is, bind a reference to it" reference.
In this case it will be a const lvalue reference, because:
1. `auto&&`
2. `const std::shared_ptr<Connection>& &&` after deducing `auto` (note: `connections_` is a `std::set`, which only has const iterators)
3. `const std::shared_ptr<Connection>&` after reference-collapsing

Normally I point out that `const auto&` or `auto&` would be more explicit and readable, but since this is old code with just a space change, I didn't want to bother @adamdebreceni with this.

Reply:
I am torn between always use `auto&&` in range-based for loops and never, it is really only useful when the iteration yields a temporary proxy object like with `std::vector<bool>`, `auto&` could be used, I wouldn't use `const auto&` as it makes me think that the connection is const whereas only the `std::shared_ptr<...>` is